### PR TITLE
Sysusers support nis

### PR DESCRIPTION
**Pull request by fbuihuu**

### CODE REVIEW

**Comment by yuwata** [comment]
lgtm.

**Comment by fbuihuu** [comment]
@yuwata the PR still misses the NIS support for the shadow password file. I wasn't sure it was needed but apparently it is, at least useradd does so.

Could you confirm this is required ?

**Comment by yuwata** [comment]
OK. I will try to check it. Let's tentatively drop 'good-to-merge/wating-for-ci' tag. (Sorry, it is already midnight of Friday in my country. It may take a few days.)

File: test/TEST-21-SYSUSERS/test.sh [hunk 1]
```diff
@@ -10,6 +10,13 @@ test_setup() {
         mkdir -p $TESTDIR/etc/sysusers.d $TESTDIR/usr/lib/sysusers.d $TESTDIR/tmp
 }
 
+perpare_testdir() {
```
Comment (dnicolodi) [inline test/TEST-21-SYSUSERS/test.sh]: `perpare` ?

**Comment by yuwata** [comment]
@fbuihuu I confirm that the commands provided by shadow-utils set the NIS entry (that is, lines start with [+-] ) at the end of the file. However, I can not find any sources that `+` or `-` lines in /etc/shadow or /etc/gshadow are valid settings. If I understand correctly, `[+-]` lines are for compat nss module, and the module does not read /etc/shadow or /etc/gshadow.

I think it is not necessary to do the same thing for shadow or gshadow. But, I do not oppose even if you want to implement that for the files.

**Comment by yuwata** [comment]
So, I think this can be merged if the typo in function name is fixed.

File: test/TEST-21-SYSUSERS/test.sh (hunk 1 above)
Comment (fbuihuu) [reply test/TEST-21-SYSUSERS/test.sh]: thanks for spotting the typo

**Comment by yuwata** [comment]
It seems all ubuntu CIs are not working or too slow. I confirm that the tests for sysusers pass in my local environment. Let's merge this.

**Comment by poettering** [comment]
Hmm, is this weird NIS entry in /etc/passwd documented anywhere? man:passwd(5) doesn't document it, unfortunately...

**Comment by poettering** [comment]
Hmm, so nsswitch.conf(5) appears to document this briefly. But it suggests that these weird entries need handling for the shadow file too? this PR doesn't do that though afaics.

See http://man7.org/linux/man-pages/man5/nsswitch.conf.5.html.

(interesting that only mentions these entries for shadow, not for gshadow...)

**Comment by fbuihuu** [comment]
> Hmm, so nsswitch.conf(5) appears to document this briefly. But it suggests that these weird entries need handling for the shadow file too? this PR doesn't do that though afaics.

No it doesn't as I wasn't sure it was needed.

> See http://man7.org/linux/man-pages/man5/nsswitch.conf.5.html.

Indeed... will send a PR that will do that shortly.

Thanks.

**Comment by fbuihuu** [comment]
PR available at https://github.com/systemd/systemd/pull/8499

**Pull request merged**